shell/platform/tizen/tizen_input_method_context.cc

```diff
@@ -191,6 +191,51 @@ bool TizenInputMethodContext::HandleEvasEventKeyUp(Evas_Event_Key_Up* event) {
       reinterpret_cast<Ecore_IMF_Event*>(&imf_event));
 }
 
+bool TizenInputMethodContext::HandleNuiEventKey(const char* device_name,
+                                                uint32_t device_class,
+                                                uint32_t device_subclass,
+                                                const char* key,
+                                                const char* string,
+                                                uint32_t modifiers,
+                                                uint32_t scan_code,
+                                                size_t timestamp,
+                                                bool is_down) {
+  Ecore_Event_Key event;
+  event.keyname = event.key = key ? key : "";
+  event.string = string ? string : "";
+  event.modifiers = modifiers;
+  event.keycode = scan_code;
+  event.timestamp = timestamp;
+  if (device_name) {
+    event.dev = ecore_device_add();
+    ecore_device_name_set(event.dev, device_name);
+    ecore_device_class_set(event.dev,
+                           static_cast<Ecore_IMF_Device_Class>(device_class));
+    ecore_device_subclass_set(
+        event.dev, static_cast<Ecore_IMF_Device_Subclass>(device_subclass));
+  }
+
+  if (is_down) {
+    Ecore_IMF_Event_Key_Down imf_event =
+        EcoreEventKeyToEcoreImfEvent<Ecore_IMF_Event_Key_Down>(&event);
+    if (event.dev) {
+      ecore_device_del(event.dev);
```

Comment:
Should a device be added and deleted every time a key is typed?

Reply:
oh... i will check it...

Reply:
I added `ecore_device_` and modified it to call `ecore device_del` when `TizenInputMethodContext` is deleted.